truffle/src/main/java/org/jruby/truffle/parser/RubyWarnings.java

```diff
@@ -147,19 +147,33 @@ public void warn(ID id, String fileName, int lineNumber, String message) {
      */
     @SuppressWarnings("deprecation")
     public void warn(ID id, String fileName, String message) {
-        if (!runtime.warningsEnabled()) return;
+        if (!runtime.warningsEnabled()) {
+            return;
+        }
 
         StringBuilder buffer = new StringBuilder(100);
 
         buffer.append(fileName).append(' ');
         buffer.append("warning: ").append(message).append('\n');
-        DebugHelpers.eval(runtime, "$stderr.write Truffle::Interop.from_java_string(message)", "message", buffer.toString());
+        DebugHelpers.eval(
+                runtime,
+                "$stderr.write Truffle::Interop.from_java_string(message)",
+                "message",
+                buffer.toString());
     }
 
     public void warn(ID id, String message) {
-        if (!runtime.warningsEnabled()) return;
+        if (!runtime.warningsEnabled()) {
+            return;
+        }
 
-        throw new UnsupportedOperationException();
+        StringBuilder buffer = new StringBuilder(100);
+        buffer.append("warning: ").append(message).append('\n');
+        DebugHelpers.eval(
+                runtime,
+                "$stderr.write Truffle::Interop.from_java_string(message)",
+                "message",
+                buffer.toString());
```

Comment:
It seems there could be some code sharing here, for instance by passing a null or `""` fileName to the method above.